lib/auth/auth_with_roles.go

```diff
@@ -1852,6 +1943,13 @@ func (a *ServerWithRoles) UpsertRole(ctx context.Context, role services.Role) er
 			"role option request_access: %v is only available in enterprise subscriptions", options.RequestAccess)
 	}
 
+	// access predicate syntax is not checked as part of normal role validation in order
+	// to allow the available namespaces to be extended without breaking compatibility with
+	// older nodes/proxies (which do not need to ever evaluate said predicates).
+	if err := services.ValidateAccessPredicates(role); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
The predicate is just a string to them.  They don't need to validate the contents of that string because they never do anything with its contents.  The auth server, on the other hand, needs to be able to evaluate the predicate.  That means that the predicate can't pull in context variables that the auth server doesn't know how pull in (e.g. if we add a new field to the review type we might start supporting `contains(request.new_field, "some-val")`).